tests/PHPStan/Analyser/data/TestDynamicReturnTypeExtensions.php

```diff
@@ -263,3 +264,23 @@ public function getTypeFromStaticMethodCall(
 		return $scope->getType(new New_($methodCall->class));
 	}
 }
+
+
+class ObjectHookedDynamicStaticMethodReturnTypeExtension implements DynamicStaticMethodReturnTypeExtension {
+
+	public function getClass(): string
+	{
+		return 'object';
```

Comment:
Magic strings like this are a no-go :) We need a new specific extension type. I'd imagine the interface can look the same, without the `getClass` method.

Reply:
Ok, how do we name it?

Or how about returning null here?

Reply:
Heh, `object::class` works, so it is not a magic string anymore :D

https://3v4l.org/rrvKH

Reply:
Implemented null version.

Reply:
The name isn't that important 😊 It can be "ClasslessDynamicMethodReturnTypeExtension".

Reply:
Classless is a good prefix! But I assume I should wait for [this](https://github.com/phpstan/phpstan-src/pull/2761#issuecomment-1823617901)